src/lib.rs

```diff
@@ -288,7 +286,7 @@ impl MapOptions {
         }
     }
 
-    pub fn style(mut self, style: String) -> MapOptions {
+    pub fn style(mut self, style: StyleOrRef) -> MapOptions {
```

Comment:
I slightly prefer providing two functions to single function with enum.

```rust
pub fn style_ref(mut self, style_ref: String);
pub fn style(mut self, style: Style);
```

Reply:
Yeah, I like your suggestion as well. Have added a new commit to the PR to bring it into line.